FindmallocMC.cmake

```diff
@@ -96,8 +93,14 @@ if(mallocMC_ROOT_DIR)
     string(REGEX MATCH "([0-9]+)" mallocMC_VERSION_PATCH
                                 ${mallocMC_VERSION_PATCH_HPP})
```

Comment:
Actually we could also use a [regex](https://github.com/Kitware/CMake/commit/5685817874e25bb1e8b112e66186eec4345c4635#diff-77f7dc8210bbb3d7f3c615816c89b297). leave it free, I will fix that in a follow-up pull request! :)